### ISSUE TITLE
Calling convention recovery

### ISSUE THREAD

**Issue by Muqi-Zou** [angr/angr#3703]
### Description

Function `strip_trailing_slashes` , with [source code](https://github.com/coreutils/gnulib/blob/08ba9aaebff69a02cbb794c6213314fd09dd5ec5/lib/stripslash.c#L31):
```
bool
strip_trailing_slashes (char *file)
{
  char *base = last_component (file);
  char *base_lim;
  bool had_slash;

  /* last_component returns "" for file system roots, but we need to turn
     "///" into "/".  */
  if (! *base)
    base = file;
  base_lim = base + base_len (base);
  had_slash = (*base_lim != '\0');
  *base_lim = '\0';
  return had_slash;
}
```
is decompiled as:
```
int strip_trailing_slashes(unsigned long long a0, unsigned long a1, unsigned long a2, unsigned long a3)
{
    unsigned long v0;  // [bp-0x18]
    unsigned long v2;  // rax
    unsigned long long v3;  // r14
    char *v4;  // rax

    v0 = v2;
    v3 = a0;
    v4 = last_component(a0);
    *((char *)((*(v4) == 0? v3 : v4) + (unsigned long long)(unsigned int)base_len((*(v4) == 0? v3 : v4)))) = 0;
    return;
}
```
Two things I noticed: 1. the number of input arguments is not the same 2. no return variable, which is not matching with the function return type.
For the return type, it looks like the disassembler of angr considers this function as a void function:
![234](https://user-images.githubusercontent.com/54459497/210157843-453a438f-3f3a-4137-bf76-400d587cdb4a.png)
Not sure whether the [commits](https://github.com/angr/angr/pull/3701) in #3698  can fix this, I will decompile again once the commit is merged into the master branch.

### Steps to reproduce the bug

binary is attached
[basename.zip](https://github.com/angr/angr/files/10328130/basename.zip)
